### DISCUSSION TITLE
Displaying old apparitions of 1P/Halley

### DISCUSSION THREAD

**Discussion by cricketingstargazer** [Stellarium/stellarium#3382] [Ideas]
I am using Stellarium to support research into old apparitions of Comet Halley. What I want to be able to do is to be able to display the comet's position in the sky and tail orientation (i.e. anti-solar direction) for any given date.

This implies extending the existing comet orbit file to include the MPC ephemerides for all past apparitions of Comet Halley and, for any given date, to the able to select the appropriate orbit file for the date to be displayed (e.g. the appropriate orbit file for the 12 BC apparition to display the comet's visibility in 12 BC).

Is this feasible?

**Comment by alex-w** [comment]
Yes, it possible, please see [this file](https://github.com/Stellarium/stellarium/blob/master/data/ssystem_1000comets.ini) as example. 

Important note: we're work for refactoring support of periodic comets - https://github.com/Stellarium/stellarium/pull/3373

**Reply by cricketingstargazer** [reply]
Thanks very much Alex. That is extremely helpful.

So, I just need to add the elements of the old apparitions to the 1000-comet file following this format and manually select the appropriate comet/orbit to display?

**Reply by gzotti** [reply]
In the SolarSystemEditor plugin you can add the whole file only, not select those comets you want. If you want to select, you must edit the ssystem_minor.ini in your data directory and copy over singular entries from the 1000comets file. 

Take note, currently some features may not work as expected. But we are working on several fixes around comets just in these weeks, the next release (23.3) will show at least better orbits, and should find again all historical apparitions of 1P/Halley.

**Comment by cricketingstargazer** [comment]
So, would you recommend waiting for 23.3, or is it better to make the manual fix now, entering the additional orbits by hand_

**Reply by gzotti** [reply]
I am not even sure how to trick the current version into working behaviour. Sometimes you must even try and rename to something like "1P/Helley". I hope when 23.3 will be out there will no longer be such issues. But if you can't wait, try renaming.

**Reply by cricketingstargazer** [reply]
Dear Georg

Just to let you know that my revised paper, including the Stellarium-generated graphics was accepted for publication. I have included  both the standard acknowledgement to Stellarium and the reference to the paper describing the software and, what I hope is a generous recognition of the invaluable help from the Stellarium team.

I should say that the graphics obtained were absolutely superb for the multiple apparitions of 1P/Halley that were simulated, right back to 12 BC.

Now, it is just a matter of waiting maybe as much as a year for the paper to be published...

Many thanks to everyone for their help.

Mark

**Reply by gzotti** [reply]
Thank you for the note. Today 23.3 has appeared, and the multiple Halleys come as "1P/Halley (1835)", "1P/Halley (1910)", "1P/Halley (1986)", ...
